RemZiQ/nodejs-aws-be
# Task 3

**What this PR do?**

- Add `product-service` with 2 handlers  `getAllProducts` and `getProductById`

_Additional scope_:

 - Async/await is used in lambda functions
- ES6 modules are used for product-service implementation
- Webpack is configured for product-service
- Lambda handlers (getProductsList, getProductsById) code is written not in 1 single module (file) and separated in codebase.
- Main error scenarios are handled by API ("Product not found" error, try catch blocks are used in lambda handlers).

**Task**

[task-3](https://github.com/rolling-scopes-school/nodejs-aws-tasks/blob/main/task3-product-api/task.md)


**How to test**

- Please check manually links:
 - [get all products](https://vzhywwsygf.execute-api.eu-west-1.amazonaws.com/dev/products)
 - [get product by id](https://vzhywwsygf.execute-api.eu-west-1.amazonaws.com/dev/products/6ecd6c6a-e77c-4f05-8af9-89ae9524cf76)
- change ID in `getProductById` by any of the presented(could be find here [get all products](https://vzhywwsygf.execute-api.eu-west-1.amazonaws.com/dev/products))
- change ID for invalid to make sure response still makes sense
- also check deployed [front-end application](https://dngh4t03yzrs0.cloudfront.net/)

**Screenshots**


![image](https://user-images.githubusercontent.com/43277312/97813357-aabc3580-1c98-11eb-963b-a44033053ed9.png)
